### PR TITLE
BB-397: integrate circuit breaker in ingestion producer

### PR DESCRIPTION
**Pull request by welansari**
- Integrate circuit breaker in ingestion producer
- Add control flow panel in ingestion producer and processor panels
- Add CircuitBreaker in Queue Populator config

**Ingestion Producer Dashboard**
ingestion_producer_dashboard

**Ingestion Processor Dashboard**
ingestion_processor_dashboard

Issue: BB-397

### CODE REVIEW

**Review by KillianG — APPROVED** [review]

**Review by tmacro — APPROVED** [review]

File: lib/queuePopulator/IngestionPopulator.js [hunk 1]
```diff
@@ -749,29 +766,55 @@ class IngestionPopulator {
...
+            return cb();
+        }
+        // skipping if location is paused
+        if (this._pausedLocations[logReader.getLocationConstraint()] !== undefined) {
+            this.log.debug('location paused, skipping', {
+                zenkoBucket: logReader.getTargetZenkoBucketName(),
+                location: logReader.getLocationConstraint(),
+            });
+            return cb();
+        }
```
Comment (francoisferrand) [inline lib/queuePopulator/IngestionPopulator.js]: does not seem related : did we not already stop ingestion when location was paused?
or is it juste an "optimization", to skip the last few entries which we may still get immediately when ingestion is paused

File: lib/queuePopulator/IngestionReader.js [hunk 2]
```diff
@@ -190,11 +190,7 @@ class IngestionReader extends LogReader {
             if (err) {
                 return done(err);
             }
-            if (hasMoreLog) {
-                // keep ingesting new log without waiting
-                return this.processLogEntries(params, done);
-            }
-            return done();
+            return done(null, hasMoreLog);
```
Comment (francoisferrand) [inline lib/queuePopulator/IngestionReader.js]: could this have an effect on performance of ingestion?

someone decided to immediately process log entries here, and according to the comment this avoids introducing a delay : has this delay been removed? or is there/was there another issue?

File: lib/queuePopulator/IngestionPopulator.js (hunk 1 above)
Comment (francoisferrand) [reply lib/queuePopulator/IngestionPopulator.js]: actually, this one is associated with another ticket (BB-399) and is the latest one : should we move it a separate, stacked PR maybe?
(or should it actually not have been pushed in this PR, and you simply need to reset `improvement/BB-397` to `bugfix/BB-399^` ?)

Comment (welansari) [reply lib/queuePopulator/IngestionPopulator.js]: yes i forgot to target the BB-399 PR, fixed that

File: lib/queuePopulator/IngestionReader.js (hunk 2 above)
Comment (welansari) [reply lib/queuePopulator/IngestionReader.js]: The behaviour should not change, there is still no delay in the initial scan, i only added pre checks before every log processing (or listing), where we now check the circuit breaker state as well as the location status.

This change belongs to BB-399 also btw

**Review by francoisferrand — APPROVED** [review]
